src/cmd/serve.rs

```diff
@@ -92,16 +92,34 @@ fn set_serve_error(msg: &'static str, e: errors::Error) {
     }
 }
 
-async fn handle_request(req: Request<Body>, mut root: PathBuf) -> Result<Response<Body>> {
+async fn handle_request(
+    req: Request<Body>,
+    mut root: PathBuf,
+    base_path: String,
+) -> Result<Response<Body>> {
+    let path_str = req.uri().path();
+    if !path_str.starts_with(&base_path) {
```

Comment:
What's the base_path?

Reply:
In the url `https://example.com/path/to/index.html`, base_path would be `/path/to/index.html`. It's derived on line 455.

Reply:
But we can't get it from the Request? (I haven't looked at hyper in a looooong time)

Reply:
Oh—good thought. We likely can. I also am not up on hyper and have generally stuck to using primitives throughout this PR.

I'm happy to get more involved in the project and do a subsequent PR to add more tests and do some refactoring along these lines, but for the moment this is working code that fixes something that's currently broken. I think the focus should be on getting #2170 closed sooner than later and we can circle back on the implementation details.